contracts/token/ERC721/ERC721.sol

```diff
@@ -386,8 +389,12 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
      * Emits an {Approval} event.
      */
     function _approve(address to, uint256 tokenId) internal virtual {
+        address owner = ownerOf(tokenId);
```

Comment:
We should note that ownerOf is already called in the public `approve`.

If we want to minimize duplicated `sload`s, we could use a construction similar to `_update`'s condition. This function is almost never used though, so optimising it is not as critical as transferFrom.